http-transport-binding.md

```diff
@@ -0,0 +1,338 @@
+the elements defined in the Cloud Events specification are to be used in
+[HTTP 1.1][RFC7230] requests and response messages.
+
+### 1.1. Conformance
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC2119][RFC2119].
+
+### 1.2. Relation to HTTP
+
+This specification does not prescribe rules constraining the use or handling of
+specific [HTTP methods][RFC7231-Section-4], and it also does not constrain the
+[HTTP target resource][RFC7230-Section-5-1] that is used for transferring or
+soliciting events.
+
+Events can be transferred with all standard or application-defined HTTP request
+methods that support payload body transfers. Events can be also be transferred
+in HTTP responses and with all HTTP status codes that permit payload body
+transfers.
+
+All examples herein that show HTTP methods, HTTP target URIs, and HTTP status
+codes are non-normative illustrations.
+
+This specification also applies equivalently to HTTP/2 ([RFC7540][RFC7540]),
+which is compatible with HTTP 1.1 semantics.
+
+### 1.3. Content Modes
+
+The specification defines two content modes for transferring events:
+*structured* and *binary*. Every compliant implementation SHOULD support both
+modes.
+
+In the *structured* content mode, event metadata attributes and event data are
+placed into the HTTP request or response body using an [event
+format](#14-event-formats).
+
+In the *binary* content mode, the value of the event `data` attribute is placed
+into the HTTP request or response body as-is, with the `contentType` attribute
+value declaring its media type; all other event attributes are mapped to HTTP
+headers.
+
+### 1.4. Event Formats
+
+Event formats, used with the *stuctured* content mode, define how an event is
+expressed in a particular data format. All implementations of this
+specification MUST support the [JSON event format][JSON-format], but MAY
+support any additional, including proprietary, formats.
+
+### 1.5. Security
+
+This specification does not introduce any new security features for HTTP, or
+mandate specific existing features to be used. This specification applies
+identically to [HTTP over TLS]([RFC2818][RFC2818]).
+
+## 2. Use of Cloud Events Attributes
+
+This specification does not further define any of the [Cloud Events][CE] event
+attributes.
+
+Two of the event attributes, `contentType` and `data` are handled specially
+and mapped onto HTTP constructs, all other attributes are transferred as
+metadata without further interpretation.
+
+This mapping is intentionally robust against changes, including the addition
+and removal of event attributes, and also accommodates vendor extensions to the
+event metadata. Any mention of event attributes other than `contentType` and
+`data` is exemplary.
+
+### 2.1. contentType Attribute
+
+The `contentType` attribute is assumed to contain a [RFC2046][RFC2046]
```

Comment:
While technically correct, this section feels a bit odd. Its so short and doesn't add anything beyond what the CE spec itself says. In fact, I don't think we should be repeating what the spec says since it could lead to things being out of sync at some point. For now I think we can remove this section

Reply:
I'm using the section as a reference point in this spec, and it's also putting a hard reference on the contentType attribute in the main spec. If there were a debate breaking out about contentType in the main spec potentially also accommodating some oder kind of reference, I'm saying here that this won't work for HTTP. So that reinforcement of the string REALLY being RFC2046 is deliberate, because I prescribe how to evaluate it.

Reply:
I'd like to keep the "contentType" section because it discusses one of the two properties that we do special handling for and with, even through I admit that it doesn't say anything substantially new for the property itself aside from aforementioned reinforcement of RFC2046.

Reply:
I'd prefer not to duplicate RFC references from the spec.  Propose simply linking to https://github.com/cloudevents/spec/blob/master/spec.md#contenttype